### ISSUE TITLE
[Bug Report] When I fire up stash it spews "controll characters are not allowed in yaml"

### ISSUE THREAD

**Issue by cjemorton** [stashapp/stash#3622]
**Describe the bug**
When stash runs it throws errors into the log and dies.

>goroutine 1 [running]:
main.handleError({0x1f27e60?, 0xc000345fc0})
        github.com/stashapp/stash/cmd/stash/main.go:53 +0x7c
main.recoverPanic()
        github.com/stashapp/stash/cmd/stash/main.go:44 +0x58
panic({0x15b1cc0, 0xc000418360})
        runtime/panic.go:884 +0x212
main.main()
        github.com/stashapp/stash/cmd/stash/main.go:24 +0x56

**To Reproduce**
1. Go to 'run stash'

**Expected behavior**
Stash to run.

**Screenshots**
![image](https://user-images.githubusercontent.com/2660596/228988000-e644581b-74f7-4427-adcc-b67b18355a12.png)

**Stash Version: (from Settings -> About):**
Latest version - it's been happening for a couple weeks. I figured it might be fixed so I never filed a bug report. The latest version came out and it's still doing it.

**Desktop (please complete the following information):**
 - OS: 13.1-RELEASE-p3 FreeBSD 13.1-RELEASE-p3 GENERIC amd64 
 - (With kldload linux64 - I'm running it in linux compatability mode - this has worked fine since I started using stash)
 - Browser [Pretty much any browser, chrome, firefox - this isn't about the browser if the program won't even run on the server.]
 
**Additional context**
I have stash running inside of a FreeBSD Jail, in linux compatability mode. - This has been working fine for months - it's never been an issue. I have a script i've written that allows me to update to the latest version of stash when I type 'up' inside the jail. It downloads the latest version of stash from github and stops the old version and starts the new one. This has been working fine for months.

Now stash won't even start. I'm not sure what other details you need to help diagnose this. Hopefully the screenshot was able to show the issue.

**Comment by WithoutPants** [comment]
You're screenshot and log excerpt doesn't contain the part that you've mentioned in the title. Can you please post the full context of the panic?

**Comment by cjemorton** [comment]
> Your screenshot and log excerpt doesn't contain the part that you've mentioned in the title. Can you please post the full context of the panic?

Yes I was just doing that. Sorry one moment.

**Comment by cjemorton** [comment]
[https://gist.github.com/cjemorton/d52498192e40106d2c16a4cdc137e19c](url)

**Comment by cjemorton** [comment]
That's a fresh file for the recent run of stash.

It will just balloon with those errors in a very short time.

It's currently running adding more lines, so that's just a snapshot at the moment I copied it.

**Comment by WithoutPants** [comment]
I think its adding more lines because your script is restarting it each time.

The most obvious explanation is that your config.yml file is corrupted somehow. Can you make sure that it looks correct?

**Comment by DogmaDragon** [comment]
You can try running YAML validation <https://onlineyamltools.com/validate-yaml>.

**Comment by cjemorton** [comment]
I pasted my yaml file into that url and it shows that its valid.

![image](https://user-images.githubusercontent.com/2660596/228993979-6f6d48f1-95f0-4617-a561-2f0e14dd1171.png)

**Comment by WithoutPants** [comment]
I would backup the existing config.yml file, then try pasting the contents into a new config.yml file. It's possible that any control characters aren't being pasted over. Failing that, you could try renaming the config.yml file altogether and start stash. You will need to go through the setup process again and set your paths correctly, but it might help diagnose the issue.

**Comment by cjemorton** [comment]
Is there a way to specify which config file is used on run?
There is currently a config.yml located in /root/.stash and I was just wanting to ensure that it was the one that was being used when the app was run.

I have tried moving it to a backup filename and starting the app service again, and nothing is created. Just errors in /var/log/stash.log

I've checked the rc file that controlls it, and it doesn't seem to list anything that points to a config file to use. Is the config just hardwired into the stash app?

![image](https://user-images.githubusercontent.com/2660596/228996099-11a16e4a-69bf-43cd-a090-ff4a36cbcb1e.png)

**Comment by cjemorton** [comment]
I've rebuilt the jail and setup everything with a fresh copy - the setup screen runs and I am able to see the configuration file locations options. That answers my previous question.

It's still not running in the origional jail however, and I have not migrated database over. Trying that next.

**Comment by WithoutPants** [comment]
> When stash is run - how does it know where to look to find a config file to use? Is it just hardwired to look in the home directory of the user that is used to run stash and look for a .stash directory, and pull the config.yml file from there?

It looks in the current working directory first. If it doesn't find one there, then it looks in `$HOME/.stash`. You can specify a config file to use with `-config` or `-c`.

**Comment by cjemorton** [comment]
I am going to close this. I did get it working.
I was unable to track down the source of the issue, however I basically did a full reinstall of my whole operating enviornment.
I built a new jail, installed stash in there - migrated the sqlite database over.

Thanks for your assistance - I still have the origional jail archived so I may at some point try to figure out what went wrong - I suspect something got corrputed somewhere.
